### PR TITLE
Docs working branch for Sourcegraph June release

### PR DESCRIPTION
**Pull request by MaedahBatool**
Hey folks, this is the docs working branch for Sourcegraph June 6th release. Please start adding any docs changes here and going @MaedahBatool for review. 

- The branch has docs for `smartContextWindow` configuration ([Preview link](https://sourcegraph-docs-git-sg-june-release-sourcegraph-f8c71130.vercel.app/cody/clients/enable-cody-enterprise#smart-context-window)) — DRI @aramaraju, @abeatrix

### CODE REVIEW

**Comment by MaedahBatool** [comment]
@aramaraju would love to get your review on the `smartContextWindow` config

**Review by aramaraju — APPROVED** [review]
Approval contingent on the minor suggestions :)

File: docs/cody/clients/install-vscode.mdx [hunk 1]
```diff
@@ -200,6 +200,35 @@ When you have both a repository and files @-mentioned, Cody will search the repo
   <source src="https://storage.googleapis.com/sourcegraph-assets/Docs/Media/context-with-chips-0624.mp4" type="video/mp4" />
 </video>
 
+### @-mention context providers with OpenCtx
+
+<Callout type="info">OpenCtx context providers in Cody is in Experimental stage. If you have feedback or questions, please visit our [support forum](https://community.sourcegraph.com/c/openctx/10).</Callout>
+
+[OpenCtx](https://openctx.org/) is an open standard for bringing contextual info about code into your dev tools. Cody Free and Pro users can use OpenCtx providers to fetch and use context from the following sources:
```
Comment (kalanchan) [inline docs/cody/clients/install-vscode.mdx:207]: let's leave a note saying Enterprise users can also use this experimental feature but with **limited** support. We don't want to lead them on and present a roadmap, but we shouldn't block on them from trying it.

@aramaraju what do you think?

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 2]
```diff
@@ -64,8 +64,47 @@ To enable Cody Analytics:
...
+
+If your site admin has disabled the `smartContextWindow` configuration and has configured the `chatModelMaxTokens` limit, Cody will now use the `chatModelMaxTokens` limit for your selected chat models. If the `@-mention` file size does not adhere to the predefined `chatModelMaxTokens` limit, you get the usual `File too large` warning.
+
+![smart-context-disabled](https://storage.googleapis.com/sourcegraph-assets/Docs/smart-context-disabled.png)
+
+A few considerations to keep in view:
+
+- For BYOK customers, there will be a financial consideration because once the number of tokens per request increases, this results in an increase in LLM usage costs
+- Cody Gateway customers will not have any financial implications as Sourcegraph is responsible for the LLM usage cost
+- For customers who wish to consume the expanded context window without upgrading their instance, they may set the `“chatModelMaxTokens: 45000”`
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply docs/cody/clients/enable-cody-enterprise.mdx:105]: It doesn't do the smart context filtering, but i think this is referring to customers who haven't upgraded yet so they don't have the `smartContextWindow` option, so `expanded context window without upgrading their instance,` sounds correct to me.

[the remark this answers is not in the document]
Comment (abeatrix) [reply docs/cody/clients/enable-cody-enterprise.mdx:105]: maybe an additional section to talk about the smart context filtering when smartContextWindow is enabled?

File: docs/cody/clients/install-vscode.mdx (hunk 1 above)
Comment (aramaraju) [reply docs/cody/clients/install-vscode.mdx:207]: Agreed!

**Pull request merged**